lib/helpers/deployments.rb

```diff
@@ -1,17 +1,23 @@
+require 'plugins/mcollective'
+
 module Deployments # rubocop:disable Style/Documentation
   def deploy(branch, deleted)
-    if settings.use_mco_ruby
-      result = mco(branch).first
+    # Currently requires puppet_webhook to be run as a non-root
+    # user with access to running the MCollective client.
+    if settings.use_mcollective
+      result = PuppetWebhook::Mcollective.new('r10k',
+                                              'deploy',
+                                              {
+                                                dtimeout: settings.discovery_timeout,
+                                                timeout: settings.client_timeout
+                                              },
+                                              settings.client_cfg,
+                                              environment: branch).run.first
       raise result.results[:statusmsg] unless result.results[:statuscode].zero?
```

Comment:
you should also check the RPC stats that the client return and check for nodes that did not respond.  You can get to that via the client.stats method.

It'll be an instance of this https://github.com/puppetlabs/marionette-collective/blob/master/lib/mcollective/rpc/stats.rb and there check `noresponsefrom`

Reply:
Good point. I've added some code for that.

@ripienaar @binford2k - could one/both of you give one more pass over before I merge? I added some code to both the plugin and the deployments helper methods based on @ripienaar's comment.